src/rust/ide/view/src/searcher.rs

```diff
@@ -36,16 +85,37 @@ struct Model {
     logger         : Logger,
     display_object : display::object::Instance,
     list           : ListView,
+    documentation  : documentation::View,
+    doc_provider   : Rc<CloneRefCell<AnyDocumentationProvider>>,
 }
 
 impl Model {
     fn new(app:&Application) -> Self {
         let logger         = Logger::new("SearcherView");
+        let scene          = app.display.scene();
         let display_object = display::object::Instance::new(&logger);
         let list           = app.new_view::<ListView>();
-        list.resize(Vector2(SEARCHER_WIDTH, 0.0));
+        let documentation  = documentation::View::new(&scene);
+        let doc_provider   = default();
+        display_object.add_child(&documentation);
         display_object.add_child(&list);
-        Self{logger,display_object,list}
+        list.set_position_x(SUGGESTION_LIST_X);
+        documentation.set_position_x(DOCUMENTATION_X);
+        Self{logger,display_object,list,documentation,doc_provider}
+    }
+
+    fn documentation_for_entry(&self, id:Option<list_view::entry::Id>) -> String {
+        if let Some(id) = id {
```

Comment:
I believe that using `map` + `unwrap_or_default` will b much more clean here

Reply:
This won't be "or_default".